sqs_workers/queue.py

```diff
@@ -85,53 +88,84 @@ def process_queue(self, shutdown_policy=NEVER_SHUTDOWN, wait_second=10):
                 )
                 break
 
-    def process_batch(self, wait_seconds=0) -> BatchProcessingResult:
+    def process_batch(self, wait_seconds: int = 0) -> BatchProcessingResult:
         """
         Process a batch of messages from the queue (10 messages at most), return
         the number of successfully processed messages, and exit
         """
-        queue = self.get_queue()
-
         if self.batching_policy.batching_enabled:
-            return self._process_messages_in_batch(queue, wait_seconds)
+            messages = self.get_raw_messages(
+                wait_seconds, self.batching_policy.batch_size
+            )
+            success = self.process_messages(messages)
+            messages_with_success = ((m, success) for m in messages)
```

Comment:
```suggestion
            successful_messages = ((m, success) for m in messages)
```

Maybe? 😄

Reply:
Aha, that would actually be a bit misleading! `messages_with_success` is a tuple of [message object] and [whether or not that message was successfully processed]. Though I suppose the fact that you had a question here means the current naming isn't too clear either, I will try to think of an improvement